pkg/apis/core/validation/validation.go

```diff
@@ -960,11 +961,13 @@ func validateFlockerVolumeSource(flocker *core.FlockerVolumeSource, fldPath *fie
 	return allErrs
 }
 
-var validDownwardAPIFieldPathExpressions = sets.NewString(
+var validVolumeDownwardAPIFieldPathExpressions = sets.NewString(
 	"metadata.name",
 	"metadata.namespace",
 	"metadata.labels",
+	"metadata.labels[]", // represents "metadata.labels" with an arbitary subscript
```

Comment:
The volume Downward API will support `metadata.annotations` and `metadata.labels` with/without subscript, while the environment will only support `metadata.annotations` and `metadata.labels` with subscript. I just use the suffix `[]` to denote the path with subscript. This seems to be the simpler way to implement.

Reply:
does this allow literally putting `annotations[]` in the API?

Reply:
No, it's just a special notation.